csrc/ir/nodes.cpp

```diff
@@ -2063,6 +2063,32 @@ void MmaOp::setMacro(MmaMacro macro) {
   attribute<MmaMacro>(ATTR_POS_MACRO) = macro;
 }
 
+std::vector<PolymorphicValue> MmaOp::evaluate(
+    const ExpressionEvaluator& ee,
+    const std::vector<PolymorphicValue>& inputs) const {
+  // MmaOp is preceded by broadcast. For aten::matmul use the non-broadcasted
+  // inputs.
+  const auto& a =
+      ee.evaluate(input(0)->definition()->input(0)).as<at::Tensor>();
+  const auto& b =
+      ee.evaluate(input(1)->definition()->input(0)).as<at::Tensor>();
+
+  NVF_CHECK(
+      a.dim() == b.dim(), "Either both or none of A and B should be batch");
+  NVF_CHECK(
+      a.dim() == 2 || a.dim() == 3,
```

Comment:
Interesting -- I didn't know our MmaOp takes 2D (without broadcasting). IIUC, your test doesn't test this case. So either remove this feature or test it.

Reply:
I don't have a test case for 3D yet, I'll add one for that.

Reply:
I'm confused. The test you added is for 3D not 2D. The `broadcast`s there make the MmaOp's inputs 3D.

Reply:
The non-broadcasted inputs can be 2D or 3D. The checks for the non-broadcasted inputs and I need to update the checks if I am using the broadcasted inputs or have them after removing those dimensions.